File: 0205-isomorphic-strings/0205-isomorphic-strings.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        var str1 = new Dictionary<char, char>();
        var str2 = new Dictionary<char, char>();
        for(int i = 0 ; i < s.Length;++i)
        {
            var a = s[i];
            var b = t[i];

            if (str1.ContainsKey(a) && str1[a] != b)
                return false;

            if (str2.ContainsKey(b) && str2[b] != a)
                return false;

            str1[a] = b;
            str2[b] = a;


        }
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
   
    bool isIsomorphic(string s, string t) {
        int n = s.size();
        unordered_map<char,char>mp1,mp2;
        for(int i = 0 ; i < n ; ++i)
        {
            if (!mp1.count(s[i]) && !mp2.count(t[i]))
            {
                mp1[s[i]] = t[i];
                mp2[t[i]] = s[i];
            } else if (mp1[s[i]] != t[i] || mp2[t[i]] != s[i])
                return false;
        }
        
        return true;
    }
};